src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health < 0) {
+      Animal.removeDead();
+    }
+  }
 }
 
+Animal.alive = [];
+Animal.removeDead = () => {
+  Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+};
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Herbivore
+      && !animal.hidden
+    ) {
+      animal.applyDamage(50);
+    }
 }
```

Comment:
There is a missing check to ensure that the `bite` method does not work if the target animal is another carnivore. According to the task description, the method should only decrease the health of a herbivore that is not hiding, and it should not work on other carnivores.